R/summarize_draws.R

```diff
@@ -0,0 +1,108 @@
+#' @title Calculate summary statistics
+#'
+#' @description Calculate summary statistics when collapsing over a certain
+#'   variable. For example can calculate summary statistics across a set of
+#'   draws, or locations, or location-years, etc.
+#'
+#' @param dt \[`data.table()`\]\cr
+#'   Data to calculate summary statistics for.
+#' @param id_cols \[`character()`\]\cr
+#'   ID columns that uniquely identify each row of `dt`.
+#' @param summarize_cols \[`character()`\]\cr
+#'   The `id_cols` that should be collapsed and to calculate summary statistics
+#'   over.
+#' @param value_col \[`character(1)`\]\cr
+#'   Value column that summary statistics should be calculated for.
+#' @param summary_fun \[`character()`\]\cr
+#'   Names of the functions that can be used to summarize a vector of values.
+#'   Default is "mean".
+#' @param probs \[`numeric()`\]\cr
+#'   Probabilities with values in `[0,1]` to be used when producing sample
+#'   quantiles with `stats::quantile()`. Default is the 2.5 and 97.5 quantiles.
+#'
+#' @return \[`data.table()`\] with `id_cols` (minus the `summarize_cols`) plus
+#'   summary statistic columns. The summary statistic columns have the same name
+#'   as each function specified in `summary_fun` and the quantiles are named
+#'   like 'q_`(probs * 100)`'.
```

Comment:
Should we include an option to name quantile columns in summary dt "lower" and "upper"?

Reply:
I think that may be a bit too custom for a generic function like this and would be removing important information from the data.table column names.